src/Body.cpp

```diff
@@ -1,81 +1,174 @@
 #include "../include/Body.hpp"
-#include "../include/Program.hpp"
 
 namespace Lexer {
-    std::ostream &operator<<(std::ostream &os, const Body &body)
-    {
+    std::ostream &operator<<(std::ostream &os, const Body &body) {
         size_t i = 0;
         bool ret = false;
-        while (body.value[i].type == TK_SEMICOLON) {
+        while (body.value[i].type == TK_EOL) {
             i++;
         }
-        for (; i < body.value.size(); i++) {
+        size_t size = body.value.size();
+        for (; i < size; i++) {
             Types type;
 
+            if (Body::encapsulate(body, size, type, os, i)) {
+                continue;
+            }
             switch (body.value[i].type) {
+                case TK_PERIOD:
+                    os << "::";
+                    break;
+                case TK_CONST:
+                    os << "const ";
+                    [[fallthrough]];
+                case TK_LET:
+                    type =
+                            JS_ANY; // TODO: change to check further the type of the variable
+                    os << TypeNames[type] << " ";
+                    break;
+                case TK_VAR:
+                    // TODO: add the variable to the list of variables that is outof the scope
+                    type =
+                            JS_ANY; // TODO: change to check further the type of the variable
+                    os << TypeNames[type] << " ";
+                    break;
+                case TK_EOL:
+                    if (i != 0 && (body.value[i - 1].type == TK_SEMICOLON || body.value[i - 1].type == TK_LBRACE)) {
+                        continue;
+                    }
+                    [[fallthrough]];
```

Comment:
what is this ?

Reply:
you can use  [[fallthrough]] to remove compile errors when you activate `-Wall -Wextra -Werror`

```c++
void f(int n)
{
    void g(), h(), i();
 
    switch (n)
    {
        case 1: 
        case 2: // for this case the code will execute case 2
            g();
            [[fallthrough]]; // there is no break so the code will continue execute the case 3 even if n is 2 ;
        case 3: // no warning on fallthrough
            h();
            break; // with the break we will exit the scope of the switch
     }
}
```